### PR TITLE
Launch Argument Additions

### PR DESCRIPTION
**Pull request by acikek**
## Additions

- **Safe Mode**. Can be enabled with `--geode:safe-mode`. Completely skips the mod loading step.
- **Disable Custom Menu**. Can be toggled on with `--geode:disable-custom-menu`.
  - Removes text below loading bar on startup
  - Removes "updating mod index" message
  - Removes Geode button in main menu
  - **Does not** disable any Geode popups/notifications from errors occurring
- `std::optional<T> parseLaunchArgument(std::string_view const name)` on both `Loader` and `Mod`. Makes it easier to parse integer, double, etc. launch arguments, and uses `matjson` for the deserialization
  - Renamed `getLaunchBool` to `getLaunchFlag` in light of this change, but that's fine since no one's using it anyways.

> **Note**: this also establishes a `kebab-case` naming convention for launch arguments.

### CODE REVIEW

File: loader/include/Geode/loader/Loader.hpp [hunk 1]
```diff
@@ -94,17 +97,40 @@ namespace geode {
...
          */
         std::optional<std::string> getLaunchArgument(std::string_view const name) const;
         /**
-         * Get a boolean launch argument. Returns whether the argument is present and its
+         * Get a launch argument flag. Returns whether the argument is present and its
          * value is exactly `true`.
          * @param name The argument name
          */
-        bool getLaunchBool(std::string_view const name) const;
+        bool getLaunchFlag(std::string_view const name) const;
```
Comment (altalk23) [inline loader/include/Geode/loader/Loader.hpp:110]: im worried about this causing undefined symbol errors in newer mods

File: loader/include/Geode/loader/Mod.hpp [hunk 2]
```diff
@@ -163,17 +168,26 @@ namespace geode {
...
          */
         std::optional<std::string> getLaunchArgument(std::string_view const name) const;
         /**
-         * Equivalent to a prefixed `Loader::getLaunchBool` call. See `Mod::getLaunchArgument`
+         * Equivalent to a prefixed `Loader::getLaunchFlag` call. See `Mod::getLaunchArgument`
          * for details about mod-specific launch arguments.
          * @param name The argument name
          */
-        bool getLaunchBool(std::string_view const name) const;
+        bool getLaunchFlag(std::string_view const name) const;
```
Comment (altalk23) [inline loader/include/Geode/loader/Mod.hpp:181]: same here

File: loader/src/loader/LoaderImpl.cpp [hunk 3]
```diff
@@ -559,6 +559,12 @@ void Loader::Impl::refreshModGraph() {
 
     m_problems.clear();
 
+    if (Loader::get()->getLaunchFlag("safe-mode")) {
+        log::debug("Launched in Safe Mode. Skipping mod loading...");
+        m_loadingState = LoadingState::Done;
```
Comment (altalk23) [inline loader/src/loader/LoaderImpl.cpp:564]: does this actually work sounds hacky

Comment (altalk23) [reply loader/src/loader/LoaderImpl.cpp:564]: スクリーンショット 2024-01-26 1 14 39

File: loader/include/Geode/loader/Loader.hpp (hunk 1 above)
Comment (acikek) [reply loader/include/Geode/loader/Loader.hpp:110]: We've discussed this a bit in Discord -- presumably it shouldn't affect anyone

File: loader/src/loader/LoaderImpl.cpp (hunk 3 above)
Comment (acikek) [reply loader/src/loader/LoaderImpl.cpp:564]: I've tested it to the best of my ability and it works fine

File: loader/include/Geode/loader/Mod.hpp (hunk 2 above)
Comment (matcool) [reply loader/include/Geode/loader/Mod.hpp:181]: this is fine because no mod uses this yet

File: loader/src/loader/LoaderImpl.cpp (hunk 3 above)
Comment (cgytrus) [reply loader/src/loader/LoaderImpl.cpp:564]: after taking a closer look at this code, looks fine to me, but i pushed a commit that slightly changed the message and added a missing log::popNest() call

**Pull request merged**